packages/react-router/index.tsx

```diff
@@ -608,8 +615,10 @@ function useRoutes_(
 
   basename = basename ? joinPaths([parentPathname, basename]) : parentPathname;
 
-  let location = useLocation() as Location;
-  let matches = React.useMemo(() => matchRoutes(routes, location, basename), [
+  let currentLocation = useLocation() as Location;
+  let usedLocation = location || currentLocation;
+
+  let matches = React.useMemo(() => matchRoutes(routes, usedLocation, basename), [
     location,
```

Comment:
Really excited about this change @MeiKatz. One additional change: You’ll want to update the dependency array for `useMemo` to reference `usedLocation` instead of `location` so that it updates when that changes:
```
let matches = React.useMemo(() => matchRoutes(routes, usedLocation, basename), [
  usedLocation,
  routes,
  basename
]);
```

Reply:
Thank you. I think I had this in but after all the edits I may have lost it.

Reply:
@MeiKatz  - perhaps you should force push an update to your PR with this ^ fix ? 
@mjackson  - I would love to see this get merged; pending your approval... I would like to start adding *easy* transitions to my routes.

Reply:
@ThomasBurleson the problem is, that I fix this pull request everytime there is a change in the code of others. And as long as nobody wants to merge my code I am unwilling to update it again and again.

@mjackson @ryanflorence @timdorr give me a sign if you want my pull request to be merged. Then I will fix it again and we can merge it. This pull request is open for too long and I want this story to have an end in the near future. Please be so kind.